examples/hardware_test.rs

```diff
@@ -1,21 +1,126 @@
 #![feature(alloc)]
 #![no_std]
-
+/// Hardware regression tests.
+/// Need P0.03 and P0.04 to be connected (on a nrf52-dk).
+#[macro_use]
 extern crate alloc;
 
 use alloc::string::String;
 use tock::console::Console;
+use tock::gpio::{GpioPinUnitialized, InputMode};
+use tock::syscalls;
 use tock::timer;
 use tock::timer::Duration;
 
+static mut STATIC: usize = 0;
+
+trait MyTrait {
+    fn do_something_with_a_console(&self, console: &mut Console);
+}
+
+impl MyTrait for usize {
+    fn do_something_with_a_console(&self, console: &mut Console) {
+        console.write(format!("trait_obj_value_usize = {}\n", &self));
+    }
+}
+
+impl MyTrait for String {
+    fn do_something_with_a_console(&self, console: &mut Console) {
+        console.write(format!("trait_obj_value_string = {}\n", &self));
+    }
+}
+
 fn main() {
     let mut console = Console::new();
     console.write(String::from("[test-results]\n"));
+
+    test_heap(&mut console);
+    test_formatting(&mut console);
+    test_static_mut(&mut console);
+
+    test_gpio(&mut console);
+
+    test_trait_objects(&mut console);
+
+    test_callbacks_and_wait_forever(&mut console);
+
+    for _ in 0.. {
```

Comment:
Is there a final "everything works" message?

Reply:
Good point.